File: lib/bindings/langs/react-native/src/gen_typescript/templates/macros.ts

{% macro arg_list(func) %}
    {%- for arg in func.arguments() -%}
        {{ arg.name()|var_name -}}
        {%- if !loop.last %}, {% endif -%}
    {%- endfor %}
{%- endmacro %}

{%- macro field_list(rec) %}
    {%- for f in rec.fields() %}
        {{ f.name()|var_name|unquote }},
    {%- endfor %}
{%- endmacro -%}

{%- macro field_list_decl(rec) %}
    {%- for f in rec.fields() %}
    {%- call docstring(f, 1, ci) %}
    {%- match f.as_type() %}
    {%- when Type::Optional { inner_type } %}
    {%- let unboxed = inner_type.as_ref() %}
    {{ f.name()|var_name }}?: {{ unboxed|type_name }}
    {%- else %}
    {{ f.name()|var_name }}: {{ f|type_name }}
    {%- endmatch %}
    {%- endfor %}
{%- endmacro -%}

{% macro arg_list_decl(func) %}
    {%- for arg in func.arguments() -%}
        {{ arg.name()|var_name }}: {{ arg|absolute_type_name }}{{- arg|default_value -}}
        {%- if !loop.last %}, {% endif -%}
    {%- endfor %}
{%- endmacro %}

{%- macro docstring(defn, indent_tabs, ci) %}
{%- match defn.docstring() %}
{%- when Some(docstring) %}
{{ docstring|docstring(indent_tabs, ci) }}
{%- else -%}
{%- endmatch %}
{%- endmacro %}

File: lib/bindings/langs/react-native/src/gen_kotlin/templates/macros.kt

{% macro arg_list(func) %}
    {%- for arg in func.arguments() -%}
        {%- match arg.as_type() -%}         
        {%- when Type::Enum { name, module_path } -%}
        {{ arg.name()|var_name|unquote|temporary }}
        {%- when Type::Optional { inner_type } -%}
        {{ arg.name()|var_name|unquote|temporary }}
        {%- when Type::Record { name, module_path } -%}
        {{ arg|type_name(ci)|var_name|unquote -}}
        {%- else -%}
        {{ arg.name()|var_name|unquote }}{{ arg|rn_convert_type(ci) -}}
        {%- endmatch -%}
        {%- if !loop.last %}, {% endif -%}
    {%- endfor %}
{%- endmacro %}

{% macro arg_list_decl(func) %}
    {%- for arg in func.arguments() -%}
    {{- arg.name()|var_name|unquote }}: {{ arg|rn_type_name(ci) -}}, {% endfor %}
{%- endmacro %}

{%- macro field_list(rec) %}
    {%- for f in rec.fields() -%}
        {{ f.name()|var_name|unquote }},
    {%- endfor %}
{%- endmacro -%}

{% macro return_value(ret_type) %}   
    {%- match ret_type %}
    {%- when Type::Enum { name, module_path } %}readableMapOf(res)
    {%- when Type::Record { name, module_path } %}readableMapOf(res)
    {%- when Type::Sequence { inner_type } %}readableArrayOf(res)
    {%- else %}res
    {%- endmatch %}
{%- endmacro %}